code/modules/surgery/organs/augments_internal.dm

```diff
@@ -143,6 +143,46 @@
 /obj/item/organ/cyberimp/brain/anti_stun/syndicate
 	syndicate_implant = TRUE
 
+
+/obj/item/organ/cyberimp/brain/linkedsurgery
+	name = "surgical serverlink brain implant"
+	desc = "A brain implant with a bluespace technology that lets you to perform an advanced surgery through your station research server."
+	slot = ORGAN_SLOT_SURGICAL_IMPLANT
+	actions_types = list(/datum/action/item_action/update_linkedsurgery)
+	var/list/advanced_surgeries = list()
+	var/static/datum/techweb/linked_techweb
+	var/updated
+
+/obj/item/organ/cyberimp/brain/linkedsurgery/Initialize()
+	. = ..()
+	if(isnull(linked_techweb))
+		linked_techweb = SSresearch.science_tech
+	updated = 0
+
+/obj/item/organ/cyberimp/brain/linkedsurgery/proc/update_surgery()
+	for(var/i in linked_techweb.researched_designs)
+		var/datum/design/surgery/D = SSresearch.techweb_design_by_id(i)
+		if(!istype(D))
+			continue
+		advanced_surgeries |= D.surgery
+
+/obj/item/organ/cyberimp/brain/linkedsurgery/proc/show_surgery_list()
+	if(updated<length(advanced_surgeries))
+		to_chat(usr, "Surgical Implant updated.")
+		updated = length(advanced_surgeries)
+	else
+		to_chat(usr, "None of new surgical programs detected.")
+
+/datum/action/item_action/update_linkedsurgery
+	name = "Update Surgical Implant"
+
+/datum/action/item_action/update_linkedsurgery/Trigger()
+	if(istype(target, /obj/item/organ/cyberimp/brain/linkedsurgery))
```

Comment:
Is this check needed? Can anything else trigger it?

Reply:
not really needed, but I just expected a rare case just in case.